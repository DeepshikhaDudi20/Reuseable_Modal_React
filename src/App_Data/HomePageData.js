export const btnText = {
  confirmationModalBtnText: "Fancy NewDay Credit card?",
  subscribeModalBtnText: "Join Newsletter",
  errorModalBtnText: "Information Message"
};

export const headerToggleThemeBtnText = {
  opusThemeBtnText: "Opus Theme",
  aquaThemeBtnText: "Aqua Theme"
};
