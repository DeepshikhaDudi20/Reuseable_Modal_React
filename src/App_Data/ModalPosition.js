export const ModalPositionX = {
  center: "center",
  right: "right",
  left: "left"
};

export const ModalPositionY = {
  center: "center",
  start: "start",
  end: "end"
};
