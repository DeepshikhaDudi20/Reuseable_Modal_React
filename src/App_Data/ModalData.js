import { ModalPositionX, ModalPositionY } from "./ModalPosition";

export const Modal_Content = {
  confirmationModal: {
    title: "Join NewDay",
    showHeader: true,
    showOverlay: true,
    positionX: ModalPositionX.center,
    positionY: ModalPositionY.center,
    padding: "20px"
  },
  subscribeModal: {
    title: "Join Our Newsletter",
    showHeader: false,
    showOverlay: true,
    positionX: ModalPositionX.center,
    positionY: ModalPositionY.center,
    padding: "20px"
  },
  infoModal: {
    title: "Information Modal",
    showHeader: false,
    showOverlay: true,
    positionX: ModalPositionX.left,
    positionY: ModalPositionY.start,
    padding: "0"
  }
};
