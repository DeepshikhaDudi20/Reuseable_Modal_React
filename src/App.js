import { useState } from "react";
import Modal from "./shared/components/Modals/Modal";
import Button from "./shared/components/Button/Button";
import { Modal_Content } from "./App_Data/ModalData";
import * as S from "./shared/components/Modals/styles";
import Header from "./shared/components/layout/Header";
import Theme from "./shared/components/Theme/Theme";
import { ThemeStore } from "./shared/contexts/ThemeStore";
import { btnText } from "../src/App_Data/HomePageData";

const App = () => {
  const [showConfimationModal, setShowConfimationModal] = useState(false);
  const [showSubscribeModal, setShowSubscribeModal] = useState(false);
  const [showErrorModal, setShowErrorModal] = useState(false);

  // Perform action on Accept Button click
  const onClickAccept = () => {
    // Redirect to different link or routes and then call modal (as part of improvement)
    setShowConfimationModal(!showConfimationModal);
  };

  // Perform action on submitting subscribe button
  const onClickSubmit = () => {
    // Implement logic here for subscribe, submit details, perform email vaildation
    // and then close modal (as part of improvement)
    setShowSubscribeModal(!showSubscribeModal);
  };

  return (
    <ThemeStore>
      <Theme>
        <Header />
        <main>
          <S.ModalButtonsContainer>
            <Button
              showModal={showConfimationModal}
              setShowModal={setShowConfimationModal}
              btnText={btnText.confirmationModalBtnText}
            />
            <Button
              showModal={showSubscribeModal}
              setShowModal={setShowSubscribeModal}
              btnText={btnText.subscribeModalBtnText}
            />
            <Button
              showModal={showErrorModal}
              setShowModal={setShowErrorModal}
              btnText={btnText.errorModalBtnText}
            />
          </S.ModalButtonsContainer>
          <Modal
            showModal={showConfimationModal}
            setShowModal={setShowConfimationModal}
            config={Modal_Content.confirmationModal}
          >
            <h1 role="region" aria-label="NewDay Offers">
              NewDay Offers
            </h1>
            <p role="region" aria-label="Fancy NewDay Credit card ?">
              Fancy NewDay Credit card ?
            </p>
            <S.ModalFooter>
              <S.ModalButtonSecondary
                role="button"
                aria-labelledby="Decline"
                data-testid="button-decline"
                onClick={() => setShowConfimationModal(!showConfimationModal)}
              >
                Decline
              </S.ModalButtonSecondary>
              <S.ModalButtonPrimary
                role="button"
                aria-labelledby="Apply"
                data-testid="button-apply"
                onClick={onClickAccept}
              >
                Apply
              </S.ModalButtonPrimary>
            </S.ModalFooter>
          </Modal>

          <Modal
            showModal={showSubscribeModal}
            setShowModal={setShowSubscribeModal}
            config={Modal_Content.subscribeModal}
          >
            <h1 role="region" aria-label="Don't miss out!">
              Don't miss out!
            </h1>
            <p
              role="region"
              aria-label="Sign up to get the best deals, first looks and more !"
            >
              Sign up to get the best deals, first looks and more !
            </p>
            <input type="email" placeholder="Please enter your email" />
            <S.ModalFooter>
              <S.ModalButtonPrimary
                role="button"
                aria-labelledby="Count me in"
                data-testid="button-subscribe"
                type="submit"
                onClick={onClickSubmit}
              >
                Count me in
              </S.ModalButtonPrimary>
            </S.ModalFooter>
          </Modal>

          <Modal
            showModal={showErrorModal}
            setShowModal={setShowErrorModal}
            config={Modal_Content.infoModal}
            data-testid="error"
          >
            <h1 role="region" aria-label="Information Message">
              Information Message
            </h1>
            <p role="region" aria-label="Modal location is flexible">
              Modal location is flexible
            </p>
          </Modal>
        </main>
      </Theme>
    </ThemeStore>
  );
};

export default App;
