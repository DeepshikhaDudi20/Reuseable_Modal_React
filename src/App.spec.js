import React from "react";
import { render, fireEvent, screen } from "@testing-library/react";
import { axe, toHaveNoViolations } from "jest-axe";
import { configure } from "enzyme";
import { btnText, headerToggleThemeBtnText } from "./App_Data/HomePageData";
import Adapter from "enzyme-adapter-react-16";
import App from "./App";

configure({ adapter: new Adapter() });
expect.extend(toHaveNoViolations);

describe("The <App/> component", () => {
  const gameArray = ["R R", "R P"];
  const playerScore = -1;

  it("should calculate player score based on game Array ", () => {
    const score = calculateScore(gameArray);
    expect(score).toEqual(playerScore);
  });

  it("should renders modal on button click", async () => {
    render(<App />);
    const button = screen.getByRole("button", {
      name: btnText.subscribeModalBtnText
    });

    fireEvent.click(button);
    expect(screen.getByRole("dialog")).toBeTruthy();
  });

  it("should toggle theme on header button click", () => {
    render(<App />);
    const themeToggler = screen.getByText(
      headerToggleThemeBtnText.opusThemeBtnText
    );
    const headerBackground = screen.getByTestId("toggle-theme");
    let themeColor = window.getComputedStyle(headerBackground);
    expect(themeColor.color).toBe("rgb(0, 118, 128)");

    fireEvent.click(themeToggler);
    themeColor = window.getComputedStyle(headerBackground);

    expect(
      screen.getByText(headerToggleThemeBtnText.aquaThemeBtnText)
    ).toBeTruthy();

    expect(themeColor.color).toBe("rgb(0, 0, 166)");
  });

  it("should not have any accessibility violations", async () => {
    const { container } = render(<App />);
    expect(await axe(container)).toHaveNoViolations();
  });
});
