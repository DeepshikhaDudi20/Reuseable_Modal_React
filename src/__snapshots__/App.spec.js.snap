// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The <App/> component should renders <App/> snapshot 1`] = `
Array [
  <header
    aria-label="header"
    className="sc-pyfCe fCpyhd"
  >
    <div
      className="sc-jrcTuL kSYenU"
    >
      <button
        className="sc-kDvujY ktIIaV"
        data-testid="toggle-theme"
        onClick={[Function]}
        role="button"
      >
        <span
          data-testid="opus-theme"
        >
          Opus Theme
        </span>
      </button>
    </div>
  </header>,
  <main>
    <div
      className="sc-bcXHqe lkXRJF"
    >
      <button
        aria-labelledby="Fancy NewDay Credit card?"
        className="sc-gswNZR dvMuuv"
        data-testid="button-confirmation"
        onClick={[Function]}
        role="button"
      >
        Fancy NewDay Credit card?
      </button>
      <button
        aria-labelledby="Join Newsletter"
        className="sc-gswNZR dvMuuv"
        data-testid="button-subscribe"
        onClick={[Function]}
        role="button"
      >
        Join Newsletter
      </button>
      <button
        aria-labelledby="Internal Error"
        className="sc-gswNZR dvMuuv"
        data-testid="button-error"
        onClick={[Function]}
        role="button"
      >
        Internal Error
      </button>
    </div>
  </main>,
]
`;
